### ISSUE TITLE
`json_extract` exploding when inside `map`

### ISSUE THREAD

**Issue by cmackenzie1** [duckdb/duckdb#6259]
### What happens?

When reading from JSON (file or column of `json` type), if you do a select with a map, and the map key or value is populated using `json_extract_string`, the `map` fails with `Error: Invalid Input Error: Error in MAP creation: key list has a different size from value list (1 keys, 2 values)`

### To Reproduce

`example.json`

```json
{"name":"donald", "is_duck": true}
{"name":"cole", "is_duck": false, "extra": "hello, cole"}
```

Open DuckDB

```sql
select map(['name'], [json_extract_string(json, '$.name')]) from read_ndjson_objects('./example.json');
-- Error: Invalid Input Error: Error in MAP creation: key list has a different size from value list (1 keys, 2 values)
```

### OS:

macOS x64

### DuckDB Version:

0.7.0

### DuckDB Client:

cli

### Full Name:

Cole MacKenzie

### Affiliation:

Cloudflare

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by lnkuiper** [comment]
This has nothing to do with `json_extract` but rather with how we expect users to create a map.
```sql
select map(['name'], [range]) from range(10);
-- Error: Invalid Input Error: Error in MAP creation: key list has a different size from value list (1 keys, 10 values)
```
The lists must have the same length

**Comment by Mytherin** [comment]
This looks like a problem in how the map function handles constant vectors indeed. Both examples should work, as both have lists of length 1 for each row.

**Comment by Mytherin** [comment]
@LindsayWray could you perhaps have a look at this issue?

**Comment by cmackenzie1** [comment]
@lnkuiper is that documented somewhere else? Looking at the map type docs there is no mention of range.
https://duckdb.org/docs/sql/data_types/map

**Comment by lnkuiper** [comment]
@cmackenzie1 range is just the name of the value returned by the range table function:

```sql
select range from range(10);
```

Returns 0 up to 9. We often use this function as an example, it has nothing to do with MAP.

**Comment by cmackenzie1** [comment]
Thanks @lnkuiper - I was more so wondering about the use of a range table function and maps together. You mentioned 

> This has nothing to do with json_extract but rather with how we expect users to create a map.

Yet the docs for `map` never mention the use of a range table function for show an example like you provided above. 

Would be great to get it added to the docs for extra clarification on how to build maps!

**Comment by LindsayWray** [comment]
@cmackenzie1 
Thnx for reporting this issue! I just pushed a fix for it #6354

The range function is documented here: https://duckdb.org/docs/sql/functions/nested
